service/src/test/env_validation.ts

```diff
@@ -0,0 +1,97 @@
+import path from 'path';
+import { fileURLToPath } from 'url';
+import { archLogger } from '../utils/chalk-theme';
+import { TestSuite } from './test_suite';
+import 'dotenv/config';
+
+
+export async function runTests() {
+    archLogger.warn("\n\nVerify service outputs errors for invalid environment variables");
+
+    const cwd = path.dirname(fileURLToPath(import.meta.url));
+    const opts = { sourceFile: '../index.js', timeout: 1000 };
+
+    let testSuite: TestSuite;
+
+    const spawnTestSuite = () => new TestSuite(cwd, process.env.TEST_CONTRACTS_DIRECTORY!);
+
+    archLogger.warn("\n\n IP_ADDRESS");
+    testSuite = spawnTestSuite();
+    process.env.IP_ADDRESS = '';
+    await testSuite.expectOutput('IP_ADDRESS not set in .env', opts);
+
+    archLogger.warn("\n\n TCP_PORT");
+    testSuite = spawnTestSuite();
+    process.env.TCP_PORT = '';
+    await testSuite.expectOutput('TCP_PORT not set in .env', opts);
```

Comment:
Does the previous `process.env.IP_ADDRESS` being empty not throw an error here because `override: true` is set in the dotenv config in the `TestSuite` class?

Reply:
Yep -- because of that, `TestSuite` overwrites existing env variables with the contents in `.env` during its constructor.